src/main/java/com/iexec/sms/tee/config/GramineInternalServicesConfiguration.java

```diff
@@ -0,0 +1,27 @@
+package com.iexec.sms.tee.config;
+
+import com.iexec.common.tee.TeeEnclaveProvider;
+import com.iexec.sms.api.config.GramineServicesConfiguration;
+import com.iexec.sms.api.config.TeeAppConfiguration;
+import com.iexec.sms.api.config.TeeServicesConfiguration;
+import com.iexec.sms.tee.ConditionalOnTeeProvider;
+import lombok.Getter;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ConditionalOnTeeProvider(providers = TeeEnclaveProvider.GRAMINE)
+@Getter
+public class GramineInternalServicesConfiguration
+        extends GramineServicesConfiguration
+        implements TeeInternalServicesConfiguration {
+    public GramineInternalServicesConfiguration(
+            TeeAppConfiguration preComputeConfiguration,
```

Comment:
Is the `preCompute` bean injected here?

Reply:
Yes. Is that an issue?

Reply:
Not an issue, but it is pretty hidden that it is working thanks to the bean defined in the configuration class. If I understand well, any variable refactoring will break injection (?)

Reply:
I disagree it is hidden. `GramineInternalServicesConfiguration` is a configuration bean, so its constructor is based on other beans that are injected in it.
However, I do agree that renaming `preComputeProperties` somewhere could break injection. We may add `@Qualifier` annotations to avoid some nasty side-effects. But I'm pretty confident we don't need it because our tests - mainly SMS integration tests - detect whether the injection works or not.